File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        str_s_idx = 0
        str_t_idx = 0

        while str_s_idx < len(s) and str_t_idx < len(t):
            if s[str_s_idx] == t[str_t_idx]:
                str_s_idx += 1
            
            str_t_idx += 1
        
        return str_s_idx == len(s)

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int strSIdx = 0;
        int strTIdx = 0;

        while(strTIdx < t.length()){
            if(strSIdx < s.length() && s.charAt(strSIdx) == t.charAt(strTIdx)){
                strSIdx += 1;
            }

            strTIdx += 1;
        }

        if(strSIdx == s.length()) return true;

        return false;
    }
}